common/oatbox/filesystem/FileSystemService.php

```diff
@@ -141,19 +164,48 @@ public function registerLocalFileSystem($id, $path)
      */
     public function unregisterFileSystem($id)
     {
+        if (isset($this->filesystems[$id])) {
+            unset($this->filesystems[$id]);
+        }
         $adapters = $this->getOption(self::OPTION_ADAPTERS);
         if (isset($adapters[$id])) {
             unset($adapters[$id]);
-            if (isset($this->filesystems[$id])) {
-                unset($this->filesystems[$id]);
-            }
             $this->setOption(self::OPTION_ADAPTERS, $adapters);
             return true;
+        } elseif ($this->hasDirectory($id)) {
+            $directories = $this->getOption(self::OPTION_DIRECTORIES);
+            unset($directories[$id]);
+            $this->setOption(self::OPTION_DIRECTORIES, $directories);
+            return true;
         } else {
             return false;
         }
     }
 
+    /**
+     * Returns the configuration for an adapter
+     * @param string $id
+     * @return string[]
+     */
+    protected function getAdapterConfig($id)
+    {
+        $dirs = $this->getDirectories();
+        if (!isset($dirs[$id])) {
+            $config = [
```

Comment:
Early exit would simplify the cyclomatic complexity and improve readability by removing `elseif` and `else`.